### PR TITLE
Remove empty lines before docstrings in async functions

### PR DESCRIPTION
**Pull request by hauntsaninja**
Relates to #4043

### CODE REVIEW

File: src/black/nodes.py [hunk 1]
```diff
@@ -742,10 +742,6 @@ def is_multiline_string(leaf: Leaf) -> bool:
     return has_triple_quotes(leaf.value) and "\n" in leaf.value
 
 
-def is_funcdef(node: Node) -> bool:
-    return node.type == syms.funcdef
-
-
 def is_function_or_class(node: Node) -> bool:
```
Comment (hauntsaninja) [inline src/black/nodes.py:745]: This function is suspicious since many async defs don't have async_funcdef, but this can be subject for different PR

**Review by JelleZijlstra — APPROVED** [review]
